XRTK.Oculus/Packages/com.xrtk.oculus/Runtime/Providers/BoundarySystem/OculusBoundaryDataProvider.cs

```diff
@@ -0,0 +1,132 @@
+            return true;
+        }
+
+        #endregion  IMixedRealityBoundaryDataProvider Implementation
+
+        #region Oculus Utils
+
+        private static int cachedVector3fSize = Marshal.SizeOf(typeof(OculusApi.Vector3f));
+        private static OculusApi.OVRNativeBuffer cachedGeometryNativeBuffer = new OculusApi.OVRNativeBuffer(0);
+        private static float[] cachedGeometryManagedBuffer = new float[0];
+
+        /// <summary>
+        /// Returns an array of 3d points (in clockwise order) that define the specified boundary type.
+        /// All points are returned in local tracking space shared by tracked nodes and accessible through OVRCameraRig's trackingSpace anchor.
+        /// </summary>
+        public Vector3[] GetGeometry(OculusApi.BoundaryType boundaryType)
+        {
+            int pointsCount = 0;
+            if (OculusApi.GetBoundaryGeometry((OculusApi.BoundaryType)boundaryType, IntPtr.Zero, ref pointsCount))
+            {
+                if (pointsCount > 0)
+                {
+                    int requiredNativeBufferCapacity = pointsCount * cachedVector3fSize;
+                    if (cachedGeometryNativeBuffer.GetCapacity() < requiredNativeBufferCapacity)
+                    {
+                        cachedGeometryNativeBuffer.Reset(requiredNativeBufferCapacity);
+                    }
+
+                    int requiredManagedBufferCapacity = pointsCount * 3;
+                    if (cachedGeometryManagedBuffer.Length < requiredManagedBufferCapacity)
+                    {
+                        cachedGeometryManagedBuffer = new float[requiredManagedBufferCapacity];
+                    }
+
+                    if (OculusApi.GetBoundaryGeometry((OculusApi.BoundaryType)boundaryType, cachedGeometryNativeBuffer.GetPointer(), ref pointsCount))
+                    {
+                        Marshal.Copy(cachedGeometryNativeBuffer.GetPointer(), cachedGeometryManagedBuffer, 0, requiredManagedBufferCapacity);
+
+                        Vector3[] points = new Vector3[pointsCount];
+
+                        for (int i = 0; i < pointsCount; i++)
+                        {
+                            points[i] = new OculusApi.Vector3f()
+                            {
+                                x = cachedGeometryManagedBuffer[3 * i + 0],
+                                y = cachedGeometryManagedBuffer[3 * i + 1],
+                                z = cachedGeometryManagedBuffer[3 * i + 2],
+                            }.FromFlippedZVector3f();
+                        }
+
+                        return points;
+                    }
+                }
+            }
+
+            return new Vector3[0];
+        }
+
+        /// <summary>
+        /// Returns a vector that indicates the spatial dimensions of the specified boundary type. (x = width, y = height, z = depth)
+        /// </summary>
+        public Vector3 GetDimensions(OculusApi.BoundaryType boundaryType)
```

Comment:
@SimonDarksideJ unused?

Reply:
I can remove it, but I retained it for future use.